packages/util/src/internal.ts

```diff
@@ -27,17 +27,17 @@ import { bytesToUnprefixedHex, utf8ToBytes } from './bytes.js'
 import type { PrefixedHexString } from './types.js'
 
 /**
- * Returns a `Boolean` on whether or not the a `String` starts with '0x'
- * @param str the string input value
- * @return a boolean if it is or is not hex prefixed
- * @throws if the str input is not a string
+ * Returns a boolean on whether or not the the input starts with '0x' and if matches the optional length
+ * @param {string} value the string input value
+ * @param {number|undefined} length the optional length of the hex string in bytes
+ * @returns {boolean} Whether or not the string is a valid PrefixedHexString matching the optional length
  */
-export function isHexPrefixed(str: string): str is PrefixedHexString {
-  if (typeof str !== 'string') {
-    throw new Error(`[isHexPrefixed] input must be type 'string', received type ${typeof str}`)
-  }
+export function isHexString(value: string, length?: number): value is PrefixedHexString {
+  if (typeof value !== 'string' || !value.match(/^0x[0-9A-Fa-f]*$/)) return false
```

Comment:
It seems like this regex would consider `0x` a valid hex string. The regex could be changed to expect at least one hexadecimal digit to follow the prefix: `/^0x[0-9A-Fa-f]+$/`.

Reply:
I get your point, but I don't think we can make that call, as I've seen `0x` used in many places, and I think expecting a length of at least one hex would be a breaking change.